src/completion-provider/template-completion-provider.ts

```diff
@@ -50,36 +50,87 @@ export default class TemplateCompletionProvider {
       return [];
     }
 
+    const { root } = project;
     let completions: CompletionItem[] = [];
 
     if (isMustachePath(focusPath)) {
-      completions.push(...listComponents(project.fileIndex));
-      completions.push(...listHelpers(project.fileIndex));
+      completions.push(...listComponents(root));
+      completions.push(...listHelpers(root));
       completions.push(...emberMustacheItems);
     } else if (isBlockPath(focusPath)) {
-      completions.push(...listComponents(project.fileIndex));
+      completions.push(...listComponents(root));
       completions.push(...emberBlockItems);
     } else if (isSubExpressionPath(focusPath)) {
-      completions.push(...listHelpers(project.fileIndex));
+      completions.push(...listHelpers(root));
       completions.push(...emberSubExpressionItems);
     } else if (isLinkToTarget(focusPath)) {
-      completions.push(...listRoutes(project.fileIndex));
+      completions.push(...listRoutes(root));
     }
 
     return filter(completions, getTextPrefix(focusPath), { key: 'label' });
   }
 }
 
-function listComponents(index: FileIndex): CompletionItem[] {
-  return uniqueBy(index.files.filter(isComponent), 'name').map(toCompletionItem);
-}
-
-function listHelpers(index: FileIndex): CompletionItem[] {
-  return index.files.filter(isHelper).map(toCompletionItem);
-}
-
-function listRoutes(index: FileIndex): CompletionItem[] {
-  return index.files.filter(isRoute).map(toRouteCompletionItem);
+function listComponents(root: string): CompletionItem[] {
+  const jsPaths = walkSync(join(root, 'app', 'components'), {
+    directors: false,
+    globs: ['**/*.js']
```

Comment:
we should probably allow `*.ts` and `*.coffee` too 🤔

Reply:
I want to focus on keeping the current functionality in place for now. Will definitely add these add a later point 👍